services/src/main/java/org/keycloak/authentication/requiredactions/UpdateTotp.java

```diff
@@ -59,8 +73,16 @@ public void evaluateTriggers(RequiredActionContext context) {
 
     @Override
     public void requiredActionChallenge(RequiredActionContext context) {
+        if (context.getAuthenticationSession() != null &&
```

Comment:
Minor: Is the null check needed here? I think that not as if authenticationSession is null, it would be handled earlier in the request (and BTV. If `context.getAuthenticationSession()` is null, you would have anyway the NullPointerException on line 81 with your latest change).

Reply:
removed this check and extracted to a method